### PR TITLE
Enforce 65/72 commit message limits

### PR DESCRIPTION
**Pull request by lzap**
We could perhaps default to `55` as a safety net.

### CODE REVIEW

**Comment by domcleal** [comment]
This appears to be missing some changes from the foreman-dev thread:

1. Tomer replied suggesting making the limit 65, which you agreed with.
2. Ewoud replied suggesting log messages shouldn't be required to wrap at 72.  This could be done by ignoring code formatted lines, those indented with four spaces.

**Comment by lzap** [comment]
Sure, done!

File: github/pull_request.rb [hunk 1]
```diff
@@ -55,14 +55,22 @@ def check_commits_style
       if (commit.commit.message.lines.first =~ /\A(fixes|refs) #\d+(, ?#\d+)*(:| -) .*\Z/i) != 0
         warnings += "  * #{commit.sha} must be in the format ```fixes #redmine_number - brief description```\n"
       end
+      if commit.commit.message.lines.first.size > 65
+        warnings += "  * length of the first commit message line for #{commit.sha} exceeds 65 characters\n"
+      end
+      commit.commit.message.lines.each do |line|
+        if line.size > 72 && line =~ /^\s{4,}/
```
Comment (domcleal) [inline github/pull_request.rb]: Does this only add a warning if the line is starting four or more spaces?  Shouldn't it be the reverse?

File: github/pull_request.rb [hunk 2]
```diff
@@ -55,14 +55,22 @@ def check_commits_style
...
+          warnings += "  * commit message for #{commit.sha} is not wrapped at 72th column\n"
```
Comment (domcleal) [inline github/pull_request.rb]: 77nd

File: github/pull_request.rb (hunk 1 above)
Comment (lzap) [inline github/pull_request.rb]: > Does this only add a warning if the line is starting four or more spaces?  Shouldn't it be the reverse?

Oh, nice catch. Indeed.

-- 
Later,
 Lukas #lzap Zapletal

File: github/pull_request.rb (hunk 2 above)
Comment (lzap) [inline github/pull_request.rb]: > 77nd

Fixed.

Dressing up armor.

-- 
Later,
 Lukas #lzap Zapletal

**Comment by domcleal** [comment]
Merging, thanks @lzap.

**Pull request merged**